src/Compilers/Core/MSBuildTask/Csc.cs

```diff
@@ -202,11 +202,14 @@ protected internal override void AddResponseFileCommands(CommandLineBuilderExten
 
             // If not design time build and the globalSessionGuid property was set then add a -globalsessionguid:<guid>
             bool designTime = false;
-            if (HostObject != null)
+            if (HostObject is ICscHostObject csHost)
             {
-                var csHost = HostObject as ICscHostObject;
                 designTime = csHost.IsDesignTime();
             }
+            else if (HostObject != null)
+            {
+                throw new InvalidOperationException();
```

Comment:
Code style seems to be mixed here - when I was first implementing this, I think I managed to find a lot of examples of where `InvalidOperationException` had no message. However, doing a search over `Compilers/Core` (where this PR is) shows that (exactly!) half of `InvalidOperationException`s have arguments, and half don't.

Is there a reason for that, or would in a perfect world all of them have arguments? (i.e. should add a message here? If so, is there a style of messages I should follow? Localization concerns?)